freedomofpress/securedrop-dev-packages-lfs
# Use reprepro to automatically pull Tor packages

## Status

Ready for review

## Description of changes

Instead of maintaining and running a custom molecule playbook to pull
down Tor packages, we can use reprepro's built-in update functionality.

With this configuration, getting new Tor packages just requires:

$ gpg --import repo/conf/updates-keys/*.gpg
$ REPREPRO_BASE_DIR=repo reprepro --export=never update

And the latest Tor packages will be in repo/pool/t/tor/, where they
can be moved to core/focal and committed. For apt-test we should be
able to have CI do this regularly.

The container that runs the above commands will just need reprepro,
ca-certificates and dctrl-tools installed.

The bundled key was downloaded from <https://deb.torproject.org/torproject.org/A3C4F0F979CAA22CDBA8F512EE8CBC9E886DDD89.asc>.

Refs https://github.com/freedomofpress/securedrop/issues/6233.

## Test plan
* Verify bundled key matches one from deb.torproject.org and used elsewhere
* Spin up a plain Debian container, mount this repo in it, run `apt install reprepro ca-cerificates dctrl-tools`. Then run the two `gpg --import...` and `reprepro update` commands as written above and verify the latest Tor packages are in repo/pool/t/tor/, and these packages match the ones currently in the repo (or if a new upstream version is released, that they match those).
